catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/IOR.java

```diff
@@ -0,0 +1,334 @@
+// $RCSfile: IOR.java,v $
+// $Revision: 1.5 $
+// $Date: 2005/08/09 21:04:41 $
+// $Author: dietrick $
+//
+// **********************************************************************
+
+package org.codice.alliance.nsili.source;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.PrintWriter;
+
+/**
+ * Utility class that reads and decodes CORBA IOR files. For debugging
+ * purposes.
+ */
+public class IOR {
+
+    static boolean debug = false;
+    static boolean verbose = false;
+    byte[] hex;
+
+    IOR(byte[] rawIOR) {
+        hex = rawIOR;
+    }
+
+    /**
+     * The first four bytes are 'IOR:' The rest of the bytes are an
+     * encapsulation of an IOR encoded in hex. So we first unencode
+     * the hex, to create a new byte array that is the encapsulated
+     * IOR.
+     *
+     * Encapsulation is defined in the CORBA Spec, section 12.3.3 The
+     * first byte indicates byte order: TRUE for BigEndian, False for
+     * LittleEndian. In practice, it seems to be encoded as a 4-byte
+     * value.
+     *
+     * Next comes a type_id string. Strings are encoded as a long (4
+     * bytes) indicating length, followed by n bytes of ascii text.
+     * The nth byte is the null byte, for the benefit of C programs.
+     *
+     * Next is a sequence of profiles. Sequences are encoded as a long
+     * followed by n objects. In this case, a Profile consists of a
+     * ProfileID (long, 4 bytes) followed by a IIOP::ProfileBody.
+     *
+     * The ProfileID (see section 10.6) is either IOP or
+     * MultipleComponents. I don't know anything about
+     * MultipleComponents, as they are intended to be opaque. This
+     * program only parses IOP Profiles, although it successfully
+     * skips over Multiple Component Profiles.
+     *
+     * An IOP Profile is a ProfileBody, defined in 12.7.2 of CORBA
+     * spec. A ProfileBody consists of: A version, major and minor.
+     * Spec'ed as two chars, in practice two shorts. A host string A
+     * port (2 bytes) An object key. This is an opaque sequence of
+     * bytes for use by the CORBA implementation that produced the
+     * IOR.
+     *
+     */
+    void parse(PrintWriter out) {
+
+        // Make sure first four bytes are 'IOR:'
+        String prefix = new String(hex, 0, 4);
+        if (!prefix.equals("IOR:")) {
+            System.err.println("Invalid IOR");
+            System.err.println("The first four bytes should be: 'IOR:'");
+            System.err.println("Found: " + prefix);
+            System.exit(1);
+        }
+
+        int iorLength = (hex.length - 4) / 2;
+        byte[] ior = new byte[iorLength];
+        for (int hexIndex = 4, iorIndex = 0; hexIndex < hex.length; hexIndex += 2, iorIndex++) {
+
+            try {
+                ior[iorIndex] = (byte) ((hexByteToInt(hex[hexIndex]) << 4) + (hexByteToInt(hex[hexIndex + 1])));
+            } catch (NumberFormatException e) {
+                System.err.println("Index: " + hexIndex);
```

Comment:
If we are allowed to modify this class, I suggest using our logging vs system.out / err